topi/python/topi/nn/conv2d.py

```diff
@@ -433,13 +481,194 @@ def conv2d_NCHWc(data, kernel, stride, padding, dilation, layout, out_layout, ou
     kw = tvm.reduce_axis((0, kernel_width), name='kw')
 
     return tvm.compute(oshape, lambda n, oc_chunk, oh, ow, oc_block:
-                       tvm.sum(data_pad[n, ic//ic_bn, oh*HSTR+kh, ow*WSTR+kw,
-                                        ic%ic_bn].astype(out_dtype) *
-                               kernel[oc_chunk, ic//ic_bn, kh, kw, ic%ic_bn, oc_block],
+                       tvm.sum(data_pad[n,
+                                        ic // ic_bn,
+                                        oh * HSTR + kh * dilation_h,
+                                        ow * WSTR + kw * dilation_w,
+                                        ic % ic_bn].astype(out_dtype)
+                               * kernel[oc_chunk,
+                                        ic // ic_bn,
+                                        kh,
+                                        kw,
+                                        ic % ic_bn,
+                                        oc_block],
                                axis=[ic, kh, kw]),
                        name='conv2d_NCHWc', tag="conv2d_NCHWc")
 
 
+@tvm.target.generic_func
+def conv2d_NCHWc_int8(data, kernel, strides, padding, dilation, layout, out_layout,
+                      out_dtype='int32'):
+    """Conv2D operator for nChw[x]c layout.
```

Comment:
Later we can refactor this by adopting the way cuda conv2d manages different algorithms, such as adding "int8" template. After that we do not need another generic function. For now this is fine to me.